modules/fedimint-ln-client/src/lib.rs

```diff
@@ -157,9 +157,24 @@ pub enum LnReceiveState {
     Claimed,
 }
 
-async fn is_internal_payment(invoice: &Invoice, gateways: Vec<LightningGateway>) -> bool {
-    // Asserts that the first hop of the invoice route hints
-    // leads through a registered gateway into the current federation
+async fn invoice_has_internal_payment_markers(
+    invoice: &Invoice,
+    markers: (secp256k1::PublicKey, u64),
+) -> bool {
+    // Asserts that the invoice src_node_id and short_channel_id match known
+    // values used as internal payment markers
+    invoice
+        .route_hints()
+        .first()
```

Comment:
Is it guaranteed that the payment markers will be in the first route hint in the vector?